worker/jobs/bulkDeleteSubjectsJob.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or
+ * https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * /worker/jobs/bulkDeleteSubjectsJob.js
+ */
+const logger = require('winston');
+const subjectHelper = require('../../api/v1/helpers/nouns/subjects');
+const featureToggles = require('feature-toggles');
+const activityLogUtil = require('../../utils/activityLog');
+const publisher = require('../../realtime/redisPublisher');
+const jobLog = require('../jobLog');
+const utils = require('../../api/v1/helpers/verbs/utils');
+
+/**
+ * Delete subject by key.
+ *
+ * If the user has permission, remove subject from Postgres then from Redis
+ * via hook.
+ *
+ * @param {String} key - subject id or absolute path
+ * @param {Object} user
+ * @returns {Promise} - Indicates success or failure
+ */
+function deleteByKey(key, user) {
+  const params = {};
+  let deletedSubject;
+  params.key = { value: key };
+  return utils.findByKey(subjectHelper, params)
+    .then((subject) => utils.isWritable({ user }, subject))
+    .then((subject) => {
+      deletedSubject = subject;
+      return subject.destroy();
+    })
+    .then(() => Promise.resolve({ isFailed: false }))
+    .catch((err) => Promise.resolve({ isFailed: true, explanation: err }));
+}
+
+/**
+ * Executes bulk subject deletion.
+ *
+ * @param {Array} - List of subject keys, meaning that key is
+ *  a string either subject id or absolutePath.
+ * @param {Array} - nouns/subjects/readOnlyFields
+ * @param {Object} - user
+ * @returns {[Promise]} - Indicates success or failure for each
```

Comment:
if this is returning an array of promises, then it would be `[{Promise}]`, but also worth spelling out, e.g. `an array of Promises where each promise resolves to indicate the success or failure deleting one individual subject` (or something like that)